### PR TITLE
Add parcel-tracker extension

### PR DESCRIPTION
**Pull request by jjhok**
## Description

This extension is tracker packages by tracking number

## Screencast

track
manage

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

**Review by unnamedd — CHANGES_REQUESTED** [review]
Hello @jjhok, 
thanks for your contribution to our repository, however, giving a quick review, I noticed you used your native language (I can't say if Japanese or Chinese, sorry) to write the extension, and based on the guidelines of this repository, you must use English, due to the non-localisation support by Raycast.

Please, give a read on the page [Prepare an Extension for Store](https://developers.raycast.com/basics/prepare-an-extension-for-store), in special in the section [Localization / Language](https://developers.raycast.com/basics/prepare-an-extension-for-store#localization-language).

**Comment by jjhok** [comment]
Hello 
Thank you for the review. 🙂

I changed all the Korean guide comments to English a while ago.

However, please understand that the delivery company's name cannot be changed because it is a unique name.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @jjhok 👋 

Sorry for the ultra late response, I'm not sure, but somehow I lost track of it, I'm so sorry!

Do you have some package numbers I can try to search for?

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

File: extensions/parcel-tracker/package.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "parcel-tracker",
+  "title": "Parcel Tracker",
+  "description": "for track packages (only available for South Korea)",
```
[the remark this answers is not in the document]
Comment (jjhok) [reply extensions/parcel-tracker/package.json:5]: Thank you for your suggestion

**Comment by jjhok** [comment]
@pernielsentikaer 👋
the sample tracking item is.. 
- vendor : "한진택배"
- invoice number : 800390692776

thank you!

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
**Hi @jjhok 👋 

Thanks for the item to test with, however, it seems like it's not working for me.

50E8Ty4mUqqSCbUZ

8wy5oPgIwEyGkjyd

Another suggestion, when using “next” it would be awesome if the default text was something else than `Couldn't find your package information` – maybe something like `Insert your invoice number` – or something like that, so it's easier to understand what's happening

swQTPRdeIhqexrqu

You're also allowed to change the navigationTitle to something like `Track Package: {vendor}`

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.**

**Comment by sxn** [comment]
hi @jjhok  - any updates? I've tried the test data you provided and got the same as @pernielsentikaer. Adding some logging to see what's happening

image

I got the following:

```
00:39:17.878 800390692776 { message: { error: '유효한 키가 아닙니다.' } }
```

My Korean's unfortunately not great, so unless Google Translate is pranking me, that error seems to say `This key is not valid` - I'm guessing it's referring to the tracking key? Regardless, it looks like the error handling isn't working properly, as the code path that is getting executed assumes no error. 😅 

PS: I've taken the libery to rebase your and migrate the code to the latest version of the API, hope you don't mind. 😸

**Comment by jjhok** [comment]
@sxn hi, 👋
I'm very very late because of my main job and api server issue.
so, sorry. 

I made a commitment after revising it a little while ago.

You can test by example parcel package.
 - vendor : CJ대한통운
 - invoiceNo. : 843085731224

Apr-07-2022 11-16-14

**Comment by jjhok** [comment]
@sxn hi, 👋
I'm very very late because of my main job and api server issue.
I was so busy that I was out of my mind.
so, sorry.

I made a commitment after revising it a little while ago.

You can test by example parcel package.

vendor : CJ대한통운
invoiceNo. : 843085731224

Apr-07-2022 11-16-14

**Comment by sxn** [comment]
hi @jjhok - no worries! thanks for fixing the issue 😄

I've done a couple more changes:
- added/fixed `isLoading` behaviour to resolve some empty state flickering issues
- call `encodeURIComponent` before creating the API endpoint URLs
- added some helpful empty views to guide users
- updated the list's `navigationTitle` when a vendor is selected

Here's it in action:

https://user-images.githubusercontent.com/1155589/162186803-1463ef0f-ebe1-481c-99a5-8d6c8a67314c.mov

**Comment by jjhok** [comment]
@sxn hi!
Wow!🤩  I checked the code that you supplemented. 
Extension completion is much higher.
I will also look at the official documents carefully and improve them from now on.

I hope it's released as the current version.
Thank you so much. 👍

**Pull request merged**